runtime/src/snapshot_utils.rs

```diff
@@ -393,7 +394,7 @@ pub fn archive_snapshot_package(
 }
 
 /// Get a list of bank snapshots in a directory
-pub fn get_bank_snapshots<P>(bank_snapshots_dir: P) -> Vec<BankSnapshotInfo>
+pub fn get_bank_snapshots<P>(bank_snapshots_dir: P, creating: bool) -> Vec<BankSnapshotInfo>
```

Comment:
The `creating` flag here seems a bit strange. If there are bank snapshots in the bank snapshot dir:

```
$ ls snapshot-dir/
1/1
2/2
3/3
4/4
```

(the non-temp ones), then if `creating` is true, wouldn't the function return paths:

```
1/1.without_accounts_hash
2/2.without_accounts_hash
3/3.without_accounts_hash
4/4.without_accounts_hash
```

but they do not exist. Is that right?

I feel like this should be two (or three) separate functions. Maybe an enum?

```rust
enum BankSnapshotType {
    Pre, // without accounts hash
    Post, // with accounts hash
}
struct BankSnapshotInfo {
    slot: Slot,
    snapshot_path: PathBuf,
    type: BankSnapshotType,
}

fn get_bank_snapshots(...) -> Vec<BankSnapshotInfo> {
    // walk the bank snapshot directory and see if each file is a pre or a post, and return them all
};

fn get_bank_snapshots_pre() -> Vec<BankSnapshotInfo> {
    // pseudo code: retain the Pre's
    get_bank_snapshots().retain(|b| b.type == BankSnapshotType::Pre)
};

fn get_bank_snapshots_post() -> Vec<BankSnapshotInfo> {
    // pseudo code: retain the Post's
    get_bank_snapshots().retain(|b| b.type == BankSnapshotType::Post)
};
```